consensus/istanbul/backend/handler.go

```diff
@@ -98,28 +142,77 @@ func (sb *Backend) SetBroadcaster(broadcaster consensus.Broadcaster) {
 	sb.broadcaster = broadcaster
 }
 
-func (sb *Backend) NewChainHead() error {
+// SetP2PServer implements consensus.Handler.SetP2PServer
+func (sb *Backend) SetP2PServer(p2pserver consensus.P2PServer) {
+	sb.p2pserver = p2pserver
+	sb.valEnodeTable.p2pserver = p2pserver
+}
+
+// This function is called by worker.go whenever it gets a newWork event.
```

Comment:
Can you elaborate? Under what circumstances does worker.go get a `newWork` event?

Reply:
It's when it the miner worker get's a `newwork` event here:  https://github.com/celo-org/celo-blockchain/blob/sentry_v1/miner/worker.go#L470

Reply:
Ah yes I meant in the comment, not for me 😄